### DISCUSSION TITLE
White dots on graph/Raw Values not showing with Hardware data source Libre Bluetooth

### DISCUSSION THREAD

**Discussion by sgmoore** [NightscoutFoundation/xDrip#3378] [Q&A]
I'm not sure that I am using the correct terminology, but I can't get the graph on my new phone to look the same as the graph on my old phone and in particular it is missing the white dots as in the screenshot below

![xdrip_raw (Small) (Custom)](https://github.com/NightscoutFoundation/xDrip/assets/754367/6b0485f6-1d21-4567-ad6c-1092e9b73630)

This was using the patched Libre2 app, and was very handy in that it give me more information than just the current reading. For example, the graph above shows that even though my current glucose reading is 8.8, it is very likely to go way over 10 mmol/l

But my old phone died and my new phone is running Android 13 and I was told that you need the latest version of LibreLink to work reliably on that OS, which means I can't use the patched app any more.

The BG Readings Table shows values labelled "Raw" and "Aged raw", so I am assuming XDrip is getting the raw values but just not displaying them on the graph.

Tried on the latest Xdrip version (and also on a version built from source a few days ago) and also I am using the latest version of OOP2 (27da3f5)

I have ticked the following

Less Common/Advanced for Libre2/Show raw values in graph
xDrip+ Display Settings/Graph Settings/Display raw data plot 
xDrip+ Display Settings/Graph Settings/Show Libre trend

Am I missing something?

**Comment by Navid200** [comment]
@PhilTU

**Comment by PhilTU** [comment]
Firstly you might find the old patched app works ok, or it may fail to transfer data to xDrip. You would need a fresh sensor to find out.

Secondly the image above seems to show 5 minute blue dots (12 per hour) which are "standard" so I wonder if you had used the experimental 1 minute values (in engineering mode) or perhaps a private version with that feature ? https://github.com/NightscoutFoundation/xDrip/issues/2680

**Reply by sgmoore** [reply]
I did briefly try the old patched app, but it wouldn't install and brought up a message saying I needed to install it from the app store. (which I guessed might be a weird way of telling me I needed a newer version which I could get from the app store). 

I did have the experimental 1 minute option enabled, but the white dots were available both before and after enabling it.

Am I correct in my assumption that the white dots are the raw data and because they show in the BG Table, that XDrip is indeed getting the required information?

**Reply by PhilTU** [reply]
xDrip+ receives a value every minute via Bluetooth using OOP2 and Libre Bluetooth data source you see them arrive in the error/event log with a timestamp (as well as on OOP2 screen)

Currently I'm running a Sibionics sensor so can't fiddle around to see what the effect of different options is unfortunately.

The discussion at https://github.com/NightscoutFoundation/xDrip/discussions/2653 suggests that the effect you're looking for was an artefact of the patched app algorithm extrapolation in any case, so you may find the white dots (if available) line up nicely in between the blue ones now.

If you were asked to get Librelink from the app store you may have had an unpatched or badly patched v2.3 of Librelink-DE, as that's a standard Librelink response to being installed from an APK. My "feeling" is we had people running it on A14 but many experienced lack of data transmission.

**Reply by PhilTU** [reply]
another thought is that in xDrip+ system settings for OOP2 there are three options - no calibration, calibrate with raw or calibrate with glucose. That might influence the display of 1 minute data.

**Reply by sgmoore** [reply]
Thanks for replies.

You have given me some pointers where I can start looking to see if I can figure this out from the source etc (although I must admit I am not confident.)

I will maybe try the patched again, but will wait until time to switch sensors.

I tried the other two OOP2 calibrate options briefly, but "no calibration" showed values as ??0 and  "calibrate with glucose" showed me weird figures about half of my correct dosage.
